lib/iris/coord_categorisation.py

```diff
@@ -433,6 +435,12 @@ def add_season_membership(cube, coord, season, name="season_membership"):
     months = _months_in_season(season)
 
     def _season_membership(_, value: cftime.datetime) -> bool:
+        warnings.warn(
```

Comment:
This is an opportunity to inform Iris users in ALL cases of boolean data, since ALL cases will experience this problem.

Examples where the user will remain in the dark even after this PR is merged:

- Saving a boolean `Cube` (#3680)
- If we ever add another boolean coordinate category
- Some sort of flag on an `AncillaryVariable`. I have personally created examples like this to demonstrate Iris; e.g. a 2-dimensional `AncillaryVariable` varying over X and Y, indicating whether a cell is within a certain region (`True`) or not (`False`).

So could you move the warning and have it catch all cases of a boolean payload?

Reply:
Sorry I hadn't realised this was in `coord_categorisation` - a very immediate place for warning the user.

My suggestion was assuming this was part of the saving code. I retract my comment!